ksqldb-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadata.java

```diff
@@ -58,6 +58,8 @@
 
   void stop();
 
+  void stop(boolean resetOffsets);
```

Comment:
I think we should give this a default implementation here that just invokes  `stop()` and then only override the `stop(boolean resetOffsets)` for the binpacked queries to avoid leaking into the other query types too much